### DISCUSSION TITLE
My car isnt migrated to new app

### DISCUSSION THREAD

**Discussion by dvogt23** [DurgNomis-drol/mytoyota#341] [General]
Hey guys, first of all, thanks to all contributors. I used this python lib some time ago for my Toyota Proace City EV that is still in the MyT app and it worked before refactoring the endpoints after v0.9.3. Is that somehow possible to fetch battery state for car if its still on the "old" app?

Actually, if running v0.9.3 getting this error:
```python
(mytoyota) pi@pi:~/toyotest $ python3 simple_client.py
['europe']
Logging in...
Retrieving cars...
Cars:  [{'id': 4337562, 'vin': 'YAREZZKXZGJ0xxx', 'isNC': True, 'isOneAppMigrated': False, 'alias': 'Proace City Verso Electric', 'owner': True, 'claimedBy': 'TME-TD-DE', 'startDate': '2024-02-27T12:15:25.719Z', 'vehicleAddedOn': '2024-02-09T07:14:17.237Z', 'isEntitled': True, 'entitledBy': 'TME-TD-DE', 'entitledOn': '2024-02-27T12:15:25.719Z', 'ownerFlag': True, 'source': 'NMSC', 'horsePower': 136, 'hybrid': False, 'fuel': 'ELECTRIC', 'engine': 'Full Electric (50kWh)', 'transmissionType': 'E-Toggle', 'transmission': 'E-Toggle', 'grade': 'Family', 'modelName': 'Proace City Verso EV', 'modelCode': 'EQ', 'interiorColour': 'FY', 'exteriorColour': 'KCA ', 'imageUrl': 'https://dj3z27z47basa.cloudfront.net/b6631ec9-38f8-4f22-85d2-f8389a080824', 'deliveryCountry': 'DE', 'productionDate': '2022-11-23', 'modelDocumentId': '12345', 'productionYear': '2022', 'licensePlate': 'xxxxx', 'modelDescription': 'Proace City Verso EV'}]
Traceback (most recent call last):
  File "/home/pi/toyotest/simple_client.py", line 62, in <module>
    loop.run_until_complete(get_information())
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/pi/toyotest/simple_client.py", line 27, in get_information
    vehicle = await client.get_vehicle_status(car)
  File "/home/pi/toyotest/mytoyota/lib/python3.9/site-packages/mytoyota/client.py", line 256, in get_vehicle_status
    data = await asyncio.gather(
  File "/home/pi/toyotest/mytoyota/lib/python3.9/site-packages/mytoyota/api.py", line 70, in get_vehicle_status_endpoint
    return await self.controller.request(
  File "/home/pi/toyotest/mytoyota/lib/python3.9/site-packages/mytoyota/controller.py", line 241, in request
    raise ToyotaActionNotSupported(
mytoyota.exceptions.ToyotaActionNotSupported: Action is not supported on this vehicle
```

**Comment by CM000n** [comment]
Nope, sorry. The old MyT app is no longer supported by us.
You should actually be able to initiate the migration of your vehicle yourself when you install the new app and open it for the first time.

**Reply by dvogt23** [reply]
Unfortunately its not working, get error about: Not compatible with your car 😢  Do you have an idea, what I could do? 🙏🏾

**Reply by CM000n** [reply]
No, unfortunately not 😢  What is the year of manufacture of your Proace City EV? Which country are you in? You might want to contact your Toyota dealer about this.

**Reply by dvogt23** [reply]
Its `'productionDate': '2022-11-23'` and Im in germany. Thanks will contact my dealer 🤞🏾

**Reply by CM000n** [reply]
So newer than my own Coralla 😆  And I also live in Germany. 
As far as the year of manufacture and country are concerned, the conversion should be possible in any case.
But I have just seen that on an information page about the new MyToyota app, your vehicle is not listed among the compatible vehicles (bottom of page): https://www.toyota.de/zubehoer-service/multimedia/mytoyota-app

**Reply by dvogt23** [reply]
Ok, damn, bad news 😢 then it won't be possible, even for my dealer. 
Again: Its not possible to me to use an older version of mytoyota to get some more data from old api? Couldnt get intercept something with mitmproxy until now ... any tips?

**Reply by CM000n** [reply]
Presumably, changes were also made to the endpoints of the old API, which we did not adapt after the release of the new Mytoyota APP.
But I'm afraid you'll have to have a look for yourself and test what data is currently returned by the endpoints. At least it looks like you will still be able to authenticate yourself and receive basic information about your vehicle.
Unfortunately, I don't currently have the time to do this for the old MyT API. 😞

**Reply by dvogt23** [reply]
Thanks will take a look. Anyways it would be nice to have a link to compatible cars somewhere in description for other happy ProAce owners :(

**Reply by CM000n** [reply]
Yes, that's a good idea! Unfortunately, we don't even know which vehicles are all compatible. 😞 

The API itself does not return any information about this at any point. It is also not 100% clear from Toyota itself. On the page I have shown you, it is also not certain how well maintained it is.

At best, we could keep a list of which vehicles we think are compatible. However, this list would be rather small, as currently only I have tested or can test it with my Corolla and Gitoldgrumpy with his GR Yaris. All other compatible vehicles would then have to be filled in by the users.

**Reply by dvogt23** [reply]
Just this link would be enough though
> So newer than my own Coralla 😆 And I also live in Germany. As far as the year of manufacture and country are concerned, the conversion should be possible in any case. But I have just seen that on an information page about the new MyToyota app, your vehicle is not listed among the compatible vehicles (bottom of page): https://www.toyota.de/zubehoer-service/multimedia/mytoyota-app

**Reply by Benvandamme** [reply]
Hi, did you find any solution to gather the state of charge?  I am in the same situation and it seems there will be no solution from toyota.

**Reply by dvogt23** [reply]
My toyota is migrated now, after did contact the support etc. and it finally works with this lib

**Reply by Benvandamme** [reply]
That sounds great. Since when is it migrated and what year of production do you have? I contacted the support yesterday. Wish me luck.

**Reply by dvogt23** [reply]
Its already more than a year ago and its from 2022. Good luck 🤞🏾

**Reply by Benvandamme** [reply]
Thank you for your reply and your insights.
